api/v1beta1/azurecluster_validation.go

```diff
@@ -556,6 +557,23 @@ func validatePrivateDNSZoneName(privateDNSZoneName string, controlPlaneEnabled b
 	return allErrs
 }
 
+// validatePrivateDNSZoneResourceGroup validates the PrivateDNSZoneResourceGroup.
+func validatePrivateDNSZoneResourceGroup(privateDNSZoneName string, privateDNSZoneResourceGroup string, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if privateDNSZoneResourceGroup != "" {
+		if valid.IsNull(privateDNSZoneName) {
```

Comment:
```suggestion
		if privateDNSZoneName == "" {
```
Since privateDNSZoneName is a string